### PR TITLE
Fixes #24850 - Do not add usergroup to itself

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

**Comment by ares** [comment]
If I understand correctly, we display the same hostgroup in edit form? Could we also hide it? The check is still important for API, but we should prevent users doing wrong assignment before they hit the error.

File: app/views/usergroups/_form.html.erb [hunk 1]
```diff
@@ -21,7 +21,7 @@
         <hr>
       <% end %>
       <%= text_f f, :name %>
-      <%= multiple_checkboxes f, :usergroups, @usergroup, Usergroup,
+      <%= multiple_checkboxes f, :usergroups, @usergroup, Usergroup.except_current(@usergroup),
```
Comment (xprazak2) [inline app/views/usergroups/_form.html.erb:24]: We hide it.

**Review by ares — CHANGES_REQUESTED** [review]
Usergroup test failure seems related

**Comment by xprazak2** [comment]
[test katello]

**Comment by xprazak2** [comment]
[test katello]

**Comment by xprazak2** [comment]
Rebased and updated tests, lets see what Jenkins will say.

**Comment by xprazak2** [comment]
I do not expect content view and audit test are affected by the changes.

**Review by ares — APPROVED** [review]
Thanks @xprazak2, fixes the issue, merging!

**Pull request merged**